client/src/actions/user.constants.js

```diff
@@ -0,0 +1,33 @@
+import { userConstants } from "./constants";
+import axios from "../helpers/axios";
+
+export const signup = (user) => {
+    console.log(user);
+
+    return async (dispatch) => {
+        dispatch({ type: userConstants.USER_REGISTER_REQUEST });
+        const res = await axios.post('/signup', {
```

Comment:
You not need to create res const you can directly restructure data
ex. `const {data} = await axios...`